query_class_config.hh

```diff
@@ -40,9 +38,4 @@ inline bool operator==(const max_result_size& a, const max_result_size& b) {
     return a.soft_limit == b.soft_limit && a.hard_limit == b.hard_limit;
 }
 
-struct query_class_config {
-    reader_concurrency_semaphore& semaphore;
-    max_result_size max_memory_for_unlimited_query;
-};
```

Comment:
Good, I never liked this struct, always felt like an arbitrary pair of params, but at the time it was necessary.